### PR TITLE
PUB-1313 Stop duplicate subscriptions for a user

### PR DESCRIPTION
**Pull request by joshblackmoor**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-1313

### Change description ###

New method added called duplicateSubscriptionHandler which checks if any subscriptions match the same criteria as the new subscription being added, if there is then the existing one gets deleted and the new one supersedes it. 

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by junaidiqbalmoj — COMMENTED** [review]
Can you add unit test for it?

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionServiceImpl.java [hunk 1]
```diff
@@ -49,6 +49,8 @@ public class SubscriptionServiceImpl implements SubscriptionService {
 
     @Override
     public Subscription createSubscription(Subscription subscription) {
+        duplicateSubscriptionHandler(subscription);
```
Comment (ChrisS1512) [inline src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionServiceImpl.java:52]: Just needs some unit tests adding for this

**Review by ChrisS1512 — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionServiceImpl.java (hunk 1 above)
Comment (joshblackmoor) [reply src/main/java/uk/gov/hmcts/reform/pip/subscription/management/service/SubscriptionServiceImpl.java:52]: Done :)

**Review by junaidiqbalmoj — APPROVED** [review]
LGTM... Sonar is showing 100% coverage.

**Review by ChrisS1512 — APPROVED** [review]

**Pull request merged**